src/main/java/ru/yandex/practicum/filmorate/service/ReviewService.java

```diff
@@ -0,0 +1,185 @@
+                .timestamp(Instant.now().toEpochMilli())
+                .build());
+        return savedReview;
+
+    }
+
+    public Review updateReview(Review review) {
+        findReviewById(review.getReviewId());
+        //айди фильма и юзера не могут поменяться при обновлении
+        enrichReviewsByUseful(review);
+        Review updateReview = reviewStorage.updateReview(review);
+        log.info("Отзыв обновлён : {}", updateReview);
+        feedStorage.save(Feed.builder()
+                .userId(updateReview.getUserId())
+                .entityId(updateReview.getReviewId())
+                .eventType(EnumEventType.REVIEW)
+                .operation(EnumOperation.UPDATE)
+                .timestamp(Instant.now().toEpochMilli())
+                .build());
+        return updateReview;
+    }
+
+    public void deleteReview(Long reviewId) {
+        Review reviewById = findReviewById(reviewId);
+        feedStorage.save(Feed.builder()
+                .userId(reviewById.getUserId())
+                .entityId(reviewId)
+                .eventType(EnumEventType.REVIEW)
+                .operation(EnumOperation.REMOVE)
+                .timestamp(Instant.now().toEpochMilli())
+                .build());
+        reviewStorage.deleteReview(reviewId);
+        log.info("Отзыв id = : " + reviewId + " удалён");
+
+    }
+
+    public List<Review> findReviewsByFilmId(int filmId, Integer count) {
+        filmStorage.findFilmById(filmId);
+        List<Review> reviews = reviewStorage.findReviewsByFilmId(filmId).stream()
+                .limit(count)
+                .collect(Collectors.toList());
+        enrichReviewsByUseful(reviews);
+        return reviews;
+    }
+
+    public List<Review> findReviewsByFilmId(int filmId) {
+        filmStorage.findFilmById(filmId);
+        List<Review> reviewsByFilmId = reviewStorage.findReviewsByFilmId(filmId);
+        enrichReviewsByUseful(reviewsByFilmId);
+        return reviewsByFilmId;
+    }
+
+    public List<Review> findReviewsWithCount(Integer count) {
+        List<Review> reviews = findAllReviews().stream()
+                .limit(count)
+                .collect(Collectors.toList());
+        enrichReviewsByUseful(reviews);
+        return reviews;
+    }
+
+    public void addLikeDislike(Long reviewId, int userId, Boolean isLike) {
+        findReviewById(reviewId);
+        userService.findUserById(userId);
+        List<Integer> whoLikeReview = reviewLikeStorage.whoLikeReview(reviewId);
+        List<Integer> whoDislikeReview = reviewLikeStorage.whoDislikeReview(reviewId);
+        if (isLike) { //прилетел лайк
+            if (!whoLikeReview.contains(userId) && !whoDislikeReview.contains(userId)) { //проверяем, что от этого юзера нет лайка или дизлайка
+                reviewLikeStorage.addLikeDislike(reviewId, userId, true); //добавляем лайк
+                log.info("Пользователь id = : " + userId + " поставил like отзыву id = " + reviewId);
```

Comment:
При логировании параметров, лучше использовать заполнитель(placeholder) {}, а не конкатенацию.
Тогда код будет выглядеть более лаконично
```suggestion
                log.info("Пользователь id: {} поставил like отзыву id: {}", userId, reviewId);
```
Поправь, пожалуйста, все логирование в классе подобным образом